handlers/hash_utils.go

```diff
@@ -0,0 +1,85 @@
+package handlers
+
+import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/hex"
+	"hash/crc32"
+	"io"
+	"sync"
+)
+
+type HashInfo struct {
+	Crc32c string
+	Md5    string
+	Sha1   string
+	Sha256 string
+	Sha512 string
+}
+
+// CalculateBasicHashes generates hases
+func CalculateBasicHashes(rd io.Reader) (*HashInfo, error) {
+	crc32c := crc32.New(crc32.MakeTable(crc32.Castagnoli))
+	md5 := md5.New()
+	sha1 := sha1.New()
+	sha256 := sha256.New()
+	sha512 := sha512.New()
+
+	multiWriter := NewMultiCWriter(crc32c, md5, sha1, sha256, sha512)
```

Comment:
sorry, I thought this went in the last review but I guess it didn't go through: can this use `io.MultiWriter`? is there a reason for the custom multiwriter?

Reply:
That is my bad copy from existed code. Fixed in fa4f52bfb16054cca4f0cb9fa962ff96bbae8454